### PR TITLE
Do not leak file descriptors in backup_crontab

### PR DESCRIPTION
**Pull request by khardix**
Originally, if anything went wrong during the backup, the early return caused the crontab_file and possibly backup_file pointers to leak.

Issue found by static scanner.

### CODE REVIEW

File: src/crontab.c [hunk 1]
```diff
@@ -500,8 +500,8 @@ static int backup_crontab(const char *crontab_path) {
 	const char *env_value;
 	char backup_dir[MAX_FNAME], backup_path[MAX_FNAME];
 	int ch = '\0';
-	FILE *crontab_file;
-	FILE *backup_file;
+	FILE *crontab_file = NULL;
+	FILE *backup_file = NULL;
```
Comment (t8m) [inline src/crontab.c:504]: The initialization should not be necessary as we never call fclose with these `FILE *` variables uninitialized. Also calling fclose on NULL is afaik not allowed so please check whether this isn't the case. 

There are missing fclose calls on lines 566 and 593 though, could you please add them?

**Review by t8m — CHANGES_REQUESTED** [review]

File: src/crontab.c (hunk 1 above)
Comment (khardix) [reply src/crontab.c:504]: > Also calling fclose on NULL is afaik not allowed so please check whether this isn't the case.

TIL; I assumed that passing `NULL` resumed in no-op, but this is apparently not the case according to standard. Which also means that the cleanup section is wrong – if opening of the backup file fails, `fclose(backup_file)` is called with `backup_file == NULL`. I'll add guard checks for that.

> The initialization should not be necessary as we never call fclose with these FILE * variables uninitialized.

True; I'll drop them then.

> There are missing fclose calls on lines 566 and 593 though, could you please add them?

I'll take a look.

Comment (khardix) [reply src/crontab.c:504]: > There are missing fclose calls on lines 566 and 593 though, could you please add them?

Well, the `exit()` calls there kill the process at that point; *any* open files will be closed by the OS. Are the calls really necessary?

File: src/crontab.c [hunk 2]
```diff
@@ -594,18 +594,19 @@ static int backup_crontab(const char *crontab_path) {
...
 			putc(ch, backup_file);
 
-	(void) fclose(crontab_file);
-	(void) fclose(backup_file);
-
 	printf("Backup of %s's previous crontab saved to %s\n", User, backup_path);
 
-	return 0;
+cleanup:
+	if (backup_file) (void) fclose(backup_file);
```
Comment (t8m) [inline src/crontab.c:606]: Please use `backup_file != NULL` and put the statement on a separate line.

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by khardix** [comment]
Updated with the requested changes.

**Comment by t8m** [comment]
Merged. Thank you for your contribution.

**Pull request merged**